assignment6/CarRearranger.java

```diff
@@ -0,0 +1,160 @@
+package assignment6;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CarRearranger {
+
+    private int[] currentOrder;
+    private int[] destOrder;
+    private int[] placesByCarIndex;
+    private final int EMPTY = 0;
+
+    public CarRearranger(Integer[] sourceOrder, Integer[] destOrder) {
+        validateParameters(sourceOrder, destOrder);

```

Comment:
I like the validation you made, but I'm not sure if this is necessary or not as in the instruction is written that we don't have to validate the input...